chia/wallet/puzzles/aggregator.clvm

```diff
@@ -0,0 +1,133 @@
+        )
+    )
+
+    (defun ensure-order (MOD_HASH MOD_LOCK_HASH solutions position)
+        (if (all solutions (> 10000 position))
+            (c
+                ; create position coin to (solution + position) to lock this solution
+                (list CREATE_COIN (get-lock-puzzle-hash MOD_LOCK_HASH (sha256tree1 (f solutions)) position) 0)
+                (c
+                    ; ensure position coin is sent a correct position of this solution
+                    (list ASSERT_PUZZLE_ANNOUNCEMENT (sha256
+                                                           (get-lock-puzzle-hash MOD_LOCK_HASH (sha256tree1 (f solutions)) position)
+                                                           position)
+                                                     )
+                    (c
+                        ; announce to lock coin that we're indeed including the solution
+                        (list CREATE_COIN_ANNOUNCEMENT (concat LOCK_ANN_PREFIX (sha256tree1 (list position (sha256tree1 (f solutions))))))
+                        (ensure-order MOD_HASH MOD_LOCK_HASH (r solutions) (+ position 1))
+                    )
+                )
+            )
+            ()
+        )
+    )
+
+
+    (defun morph-and-aggregate (morphed_conditions amount solutions conditions)
+        (aggregate-solutions
+            ; create coin puzzle hash
+            (f (r (f morphed_conditions)))
+            amount
+            ; rest of solutions
+            (r solutions)
+            ; pass on the last create coin to use in the end
+            (f morphed_conditions)
+            ; merge inner conditions with existing ones and pass it on to next iteration
+            (merge-lists (r morphed_conditions) conditions)
+        )
+    )
+
+    (defun aggregate-solutions (INNER_AGG_PUZZLE_HASH AMOUNT solutions create_coin_condition conditions)
+        (if solutions
+            (if (= (sha256tree1 (f (f solutions))) INNER_AGG_PUZZLE_HASH)
+                (morph-and-aggregate
+                    (morph-conditions (a (f (f solutions)) (f (r (f solutions))))
+                            AMOUNT
+                            create_coin_condition
+                            conditions)
+                    AMOUNT
+                    solutions
+                    conditions
+                )
+                ;(list (morph-conditions (a (f (f solutions)) (f (r (f solutions)))) AMOUNT create_coin_condition conditions)); solutions conditions)
+                (x "bad inner puzzle")
+            )
+            ; return back create coin that will be used to curry new puzzle hash for aggregator
+            (c create_coin_condition conditions)
+        )
+    )
+
+    (defun morph-conditions (conditions singleton_amount odd_create_coin morphed_conditions)
+        (if conditions
+            (if (all (= (f (f conditions)) CREATE_COIN)); (= (f (r (r (f conditions)))) singleton_amount) (not odd_create_coin))
```

Comment:
For conditions like announcements that only have one parameter, this will raise a `path into atom` error because of the eager evaluation (announcments don't have a `(f (r (r`)